### PR TITLE
chore(node/buffer): convert to js private variables

### PR DESCRIPTION
**Pull request by sno2**
See #1649 [please wait for CI]

### CODE REVIEW

**Comment by sno2** [comment]
I'm sorry @bartlomieju , but I can't seem to find out why the tests are hanging.  All I did was change the `BufferReader` abstract class to not unsafe anymore and rename all "private" variable / method instances to `#__`.

**Comment by kt3k** [comment]
@sno2 
This seems crashing even without rust panic (maybe it crashes inside of V8?). The crash seems happening around `#reset` method of `BufReader`, but I couldn't figure out what exactly causes this.

If I renamed `#reset` (of BufReader) to `_reset`, then the issue disappeared on my local machine.

**Comment by sno2** [comment]
> @sno2 This seems crashing even without rust panic (maybe it crashes inside of V8?). The crash seems happening around `#reset` method of `BufReader`, but I couldn't figure out what exactly causes this.
> 
> If I renamed `#reset` (of BufReader) to `_reset`, then the issue disappeared on my local machine.

Yeah, ummm when I changed `#reset` to an arrow function it passed.  I am so confused because I have no idea why it would behave differently.  We'll see what CI says, though.

File: io/buffer.ts [hunk 1]
```diff
@@ -293,40 +293,40 @@ export class BufReader implements Reader {
...
   }
 
   buffered(): number {
-    return this.w - this.r;
+    return this.#w - this.#r;
   }
 
   // Reads a new chunk into the buffer.
-  private async _fill() {
+  async #fill() {
```
Comment (kt3k) [inline io/buffer.ts:309]: Regarding the case of `#reset`, I suspect this might also cause the same issue..

**Comment by kt3k** [comment]
Otherwise looks good to me

**Review by kt3k — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**